biosppy/features/eda.py

```diff
@@ -0,0 +1,229 @@
+import numpy as np
+from .. import utils
+from .. import tools as st
+
+
+def eda_param(signal, min_amplitude=0.08, filt=True, size=1., sampling_rate= 1000.):
+    """ Returns characteristic EDA events.
+
+    Parameters
+    ----------
+    signal : array
+        Input signal.
+    min_amplitude : float, optional
+        Minimum treshold by which to exclude SCRs.
+    filt: bool
+        If to filter signal to remove noise and low amplitude events.
+    size: float
+        Size of the filter in seconds
+    sampling_rate: float
+        Data acquisition sampling rate.
+
+    Returns
+    -------
+    onsets : array
+        Indices of the SCR onsets.
+    peaks : array
+        Indices of the SRC peaks.
+    amplitudes : array
+        SCR pulse amplitudes.
+    end : array
+        Indices of the SCR end.
+    """
+    
+    assert len(signal) > 1, "len signal <1"
+    signal = np.array(signal).astype(np.float)
+
+    # smooth
+    if filt:
+        try:
+            if sampling_rate > 1:
+                signal, _, _ = st.filter_signal(signal=signal,
+                                         ftype='butter',
+                                         band='lowpass',
+                                         order=4,
+                                         frequency=2,
+                                         sampling_rate=sampling_rate)
+        except Exception as e:
+            print(e, "Error filtering EDA")
+
+        # smooth
+        try:
+            sm_size = int(size * sampling_rate)
+            signal, _ = st.smoother(signal=signal,
+                                      kernel='boxzen',
+                                      size=sm_size,
+                                      mirror=True)
+        except Exception as e: print(e)
+
+
+    amps, onsets, pks, end = [], [], [], []
+    zeros = st.find_extrema(signal=signal, mode='min')[0] # get zeros
+    for z in range(len(zeros)):
+        if z == len(zeros) -1:  # last zero
+            s = signal[zeros[z]:]  # signal amplitude between event
+        else:
+            s = signal[zeros[z]:zeros[z + 1]]  # signal amplitude between event
+            
+        pk = st.find_extrema(signal=s, mode='max')[0]  # get pk between events
+        #print("_pks", pks)
+        for p in pk:
+            if (s[p] - s[0]) > min_amplitude:  # only count events with high amplitude
+                pks += [zeros[z] + p]
+                onsets += [zeros[z]]
+                amps += [s[p] - s[0]]
+                if z == len(zeros) -1:  # last zero
+                    end += [len(signal)]                    
+                else:
+                    end += [zeros[z + 1]]
+
+    args, names = [], []
+    names += ["onsets", "peaks", "amplitudes", "end"]
+    args += [onsets, pks, amps, end]
+
+    args = np.nan_to_num(args)
+    return utils.ReturnTuple(tuple(args), tuple(names))
+
+
+def rec_times(signal, onsets, pks):
```

Comment:
To maintain the same notation.

```suggestion
def rec_times(signal, onsets, peaks):
```